lib/Dialect/Struct/IR/Ops.cpp

```diff
@@ -270,59 +378,20 @@ LogicalResult StructDefOp::verifyRegions() {
         }
       }
     }
-    if (!foundCompute.has_value()) {
-      return msgOneFunction(emitError, FUNC_NAME_COMPUTE);
+    if (!foundProduct.has_value() && !foundCompute.has_value()) {
+      return structFunDefError(getOperation())
+             << "missing \"@" << FUNC_NAME_COMPUTE << "\" or \"@" << FUNC_NAME_PRODUCT << "\"";
```

Comment:
```suggestion
             << "missing \"@" << FUNC_NAME_COMPUTE << "\" and \"@" << FUNC_NAME_PRODUCT << "\"";
```

Reply:
I think this is subsumed by Ian's later comment, which I prefer.